### ISSUE TITLE
Show system-library in UI

### ISSUE THREAD

**Issue by cjellick** [rancher/rancher#22771]
In this bug, we hid the system-library: https://github.com/rancher/rancher/issues/22490

We want to show it for the following reasons:
- if users choose to manage their own mirror for system-charts, they still can. So, it should show in the UI for them
- If something breaks with the system-chart (like cant reach internet), we want to be able to surface that in the UI.

In addition to showing the system-charts catalog, we also want to do the following:
1. At the API level, prevent the deletion of the system-chart. This should do two things:
  - remove the delete link from the system chart catalog so the UI automatically knows it cannot be deleted
  - in a validator or a store, throw a validation error if the user still attempts to delete it (ie via `curl -X DELETE https:///.../system-chart
2. If the user has set the new environment variable that causes rancher to use the embedded version of the chart, then prevent editing the system-chart. This should be done similar to 1 above.

Once this is done, we'll also need to revert the UI PR that hid the chart AND ensure that the UI logic does automatically respect the absence of the delete and edit links so that the user doesnt even see those options

**Comment by vincent99** [comment]
It was hidden because it made absolutely no sense showing up as a custom catalog, and decided nobody needed to turn it off in general because 2.3 bundles the charts with the image anyway.  If you want to show it then it needs its own dedicated control at the top like the other 3 special ones (and it's going to be weird that you specifically can't delete that one, when that is the normal method of overriding the other 3).

**Comment by cjellick** [comment]
Just a couple small notes, though i dont want this to devolve to a GitHub comment argument. I just want to straighten out some things you said:

> ...and decided nobody needed to turn it off in general because 2.3 bundles the charts with the image anyway.

This is opt-in by setting an env-var when spinning up rancher. It is not the default functionality. The default functionality (being able to view and edit) the catalog is the same v2.2

> you don't even want to be able to edit it?

Only when you are in the opt-in env-var mode.

**Comment by cjellick** [comment]
We've merged the backend to basically protect the system-chart catalog from editing or deletion. Logic is this:
- If its the system-chart catalog, we remove the "remove" link from the API and will raise an error if you send an http DELETE to that catalog's URL
- If its the system-cahrt catalog _and_ "bundled" via our new environment variable (used for airgap setups), we don't let you edit it in the same fashion (remove the "update" link and raise an error if you send a PUT)

So, we could do that backend regardless of whether we do the frontend change. I'm kicking over to team/ui to start the ~argument~ healthy debate on whether we should show in UI again.

**Comment by cjellick** [comment]
I've opened a separate issue for testing the API change

**Comment by vincent99** [comment]
So none of that was true at the time it was specifically discussed and decided to hide it, which is all I was trying to point out.  Just reverting and going back to the old behavior is not acceptable, it was grossly misleading making the single most important catalog look like a user-defined custom one.

The behavior being conditionally editable/deletable makes adding another toggle like I initially suggested weird.

So we're going with eliminating the special controls at the top entirely and making it a single list of all the catalogs with no specific distinction of which ones are "custom".  If you're looking at the global level then you get an extra grayed out row for each of the standard 4 catalogs (if they are not already enabled).

I also made the URLs not links since they are generally to a git server and look broken if you click them even though it's fine.

![image](https://user-images.githubusercontent.com/753917/65265978-fb89c280-dac6-11e9-8cc7-ff514a426969.png)

**Comment by jiaqiluo** [comment]
The bug fix is validated in Rancher:v2.3.0-rc6 single install

When Rancher is started **without** the `bundled` option (it is the default), the `system-library` and `library` catalogs show in the UI and are enabled, and there is no `delete` option for  the `system-library`

<img width="1481" alt="Screen Shot 2019-09-19 at 5 01 27 PM" src="https://user-images.githubusercontent.com/6218999/65289444-84245500-daff-11e9-94a9-8a51bd6ad56a.png">

When Rancher is started **with** the `bundled` option, the `system-library` and `library` catalogs still show in the UI but `library` is disabled (it can be enabled), and there are no `delete` or `edit` options for  the `system-library`

<img width="1451" alt="Screen Shot 2019-09-19 at 4 29 01 PM" src="https://user-images.githubusercontent.com/6218999/65289499-be8df200-daff-11e9-87b9-668a5c1143ea.png">
